Sources/S5TF/utils/S5TFUtils.swift

```diff
@@ -35,4 +35,82 @@ public struct S5TFUtils {
 
         return (output, task.terminationStatus)
     }
+
+    /// Extract a downloaded archive
+    /// 
+    /// - Parameters:
+    ///   - `archiveURL`: the URL for the archive.
+    ///
+    /// - Returns: output, termination status of the command.
+    ///
+    /// - Usage Example:
+    ///   - Extract "archive.tgz":
+    ///     ```
+    ///     extract(archiveURL: URL(string: "archive.tgz")!, fileExtension: "tgz")
+    ///     ```
+    ///   - Extract "another_archive.gz"
+    ///     ```
+    ///     extract(archiveURL: URL(string: "archive_archive.gz")!, fileExtension: "gz")
+    ///     ```
+    ///
+    @discardableResult
+    static public func extract(archiveURL: URL, fileExtension: String) throws -> (out: String?, status: Int32) {
```

Comment:
Is the file extension also included in the `archiveURL`? If so we could use `archiveURL.pathExtension`.

---

This is a matter of personal preference, but

```swift
static public func extract(fileAtURL archiveURL: URL, fileExtension: String) throws -> (out: String?, status: Int32) {
```

would be a little more swifty.

---

What does `out` entail? It would be nice to return the path/URL where the file got extracted, if possible.